google/cloud/logging_v2/handlers/app_engine.py

```diff
@@ -114,11 +114,14 @@ def emit(self, record):
         """
         message = super(AppEngineHandler, self).format(record)
         gae_labels = self.get_gae_labels()
-        trace_id = (
-            "projects/%s/traces/%s" % (self.project_id, gae_labels[_TRACE_ID_LABEL])
-            if _TRACE_ID_LABEL in gae_labels
-            else None
-        )
+        http_request, trace_id = get_request_data()
+        if trace_id is not None:
+            trace_id = f"projects/{self.project_id}/traces/{trace_id}"
         self.transport.send(
-            record, message, resource=self.resource, labels=gae_labels, trace=trace_id
+            record,
+            message,
+            resource=self.resource,
+            labels=gae_labels,
+            trace=trace_id,
+            http_request=http_request,
```

Comment:
I see you're using `http_pb2_request` does that work properly with the `LogEntry`?https://github.com/googleapis/python-logging/blob/6349b899811cbb16f5548df0b77564b46666c4e7/google/cloud/logging_v2/entries.py#L173

Reply:
I believe it should. This is the type used by the underlying LogEntry proto, so I assumed it would be preferred: https://github.com/googleapis/python-logging/blob/ff90fd2fb54c612fe6ab29708a2d5d984f60dea7/google/cloud/logging_v2/types/log_entry.py#L192

But I'm still getting used to working with gapic code. Do you prefer to just use dictionaries for these kinds of fields?